File: board-room/src/main/java/net/stzups/board/room/server/websocket/protocol/client/messages/ClientMessageHandshake.java
package net.stzups.board.room.server.websocket.protocol.client.messages;

import io.netty.buffer.ByteBuf;
import net.stzups.board.room.server.websocket.protocol.client.ClientMessage;
import net.stzups.board.room.server.websocket.protocol.client.ClientMessageType;

public class ClientMessageHandshake extends ClientMessage {
    private String code;

    public ClientMessageHandshake(ByteBuf byteBuf) {
        super(ClientMessageType.HANDSHAKE);
        this.code = readString(byteBuf);
    }

    public String getCode() {
        return code;
    }
}

File: scribbleshare-app/scribbleshare-frontend/src/scripts/protocol/client/messages/ClientMessageHandshake.ts
import ClientMessageType from "../ClientMessageType.js";
import ClientMessage from "../ClientMessage.js";
import ByteBuffer from "../../ByteBuffer";

export default class ClientMessageHandshake extends ClientMessage {
    invite: string;

    constructor(invite: string) {
        super(ClientMessageType.HANDSHAKE);
        this.invite = invite;
    }

    serialize(byteBuffer: ByteBuffer) {
        super.serialize(byteBuffer);
        byteBuffer.writeString8(this.invite);
    }
}